### PR TITLE
Validate duplicate axes in symbolic shape inference

### PR DESCRIPTION
**Pull request by uczltw6**
## Description

Symbolic shape inference accepted axis tuples that became duplicates after canonicalization. For example, `keras.ops.sum(x, axis=(1, -2))` on a rank-3 symbolic tensor treated both entries as axis 1, while eager backends reject the duplicate. `expand_dims(x, axis=(1, -3))` could instead leak `StopIteration` from shape construction.

This change validates uniqueness after canonicalizing axes in the shared reduction and `expand_dims` shape helpers. It raises a clear `ValueError` and keeps symbolic behavior aligned with eager execution. Two public-operation regressions cover both paths.

AI assistance was used to identify and test the edge cases. I reviewed the complete change and verified it locally.

## Validation

- focused regression tests: 2 passed
- `operation_utils_test.py` and `numpy_test.py`: 5,540 passed, 705 skipped
- Ruff check and format check on changed files
- `git diff --check`

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/operation_utils.py [hunk 1]
```diff
@@ -471,6 +475,10 @@ def reduce_shape(shape, axis=None, keepdims=False):
         axis = (axis,)
 
     axis = tuple(canonicalize_axis(a, len(shape)) for a in axis)
+    if len(set(axis)) != len(axis):
+        raise ValueError(
+            f"`axis` must not contain duplicate entries. Received: axis={axis}."
+        )
```
[the remark this answers is not in the document]
Comment (uczltw6) [reply keras/src/ops/operation_utils.py:487]: Updated in b83c22b. The reduction error now reports the original axis, its canonicalized form, and the input rank. The focused regression and all 25 `operation_utils` tests pass.

File: keras/src/ops/operation_utils.py [hunk 2]
```diff
@@ -70,6 +70,10 @@ def compute_expand_dims_output_shape(input_shape, axis):
     axis = to_tuple_or_list(axis)
     out_ndim = len(axis) + len(input_shape)
     axis = [canonicalize_axis(a, out_ndim) for a in axis]
+    if len(set(axis)) != len(axis):
+        raise ValueError(
+            f"`axis` must not contain duplicate entries. Received: axis={axis}."
+        )
```
[the remark this answers is not in the document]
Comment (uczltw6) [reply keras/src/ops/operation_utils.py:79]: Updated in b83c22b. The `expand_dims` error now reports the original axis, its canonicalized form, and the output rank. The focused regression and all 25 `operation_utils` tests pass.

**Comment by keerthanakadiri** [comment]
Hi @uczltw6 , Can you kindly sign the CLA? Thank you!
